src/libcrun/cgroup.c

```diff
@@ -1524,6 +1547,21 @@ libcrun_cgroup_enter (struct libcrun_cgroup_args *args, libcrun_error_t *err)
   int rootless;
   int ret;
 
+  /* If the cgroup configuration is limiting what CPUs/memory Nodes are available for the container,
+     then stop the container process during the cgroup configuration to avoid it being rescheduled on
+     a CPU that is not allowed.  This extra step is required for setting up the sub cgroup with the
+     systemd driver.  The alternative would be to temporarily setup the cpus/mems using d-bus.
+  */
+  if (must_stop_proc (args->resources))
```

Comment:
Are the processes started afterwards?

Reply:
yes, it is continued through the `cleanup_sig_contp` function at cleanup.

I've marked the PR as draft as I am still not 100% sure about the current fix, or whether it should be using the systemd dbus api for setting these resources and reverting them afterwards once the sub-cgroup is created.